lib/jarvis/commands/publish.rb

```diff
@@ -69,29 +69,36 @@ def execute
       if check_build?
         workdir_sha1 = git.revparse("HEAD")
         context[:sha1] = workdir_sha1
-        statuses = Octokit.statuses(project.path, workdir_sha1)
 
-        current_states = Hash[statuses.group_by(&:context).map do |ctx, info|
-           last_info = info.sort_by(&:created_at).last
-           [last_info.context, last_info.state]
-        end]
-
-        if current_states.any?
-          unsuccessful_statuses = Hash[current_states.select {|context,state| state != "success" }]
-          if unsuccessful_statuses.any?
-            logger.error("Cannot publish, some github commit hooks are not in a successful state.", :statuses => unsuccessful_statuses)
-            next
+        travis_check_run = nil
+        response = Octokit.check_runs_for_ref(project.path, workdir_sha1, filter: 'latest') # status: 'completed'
+        if response.total_count > 0
```

Comment:
maybe can get rid of this external conditional? as `check_runs.find` in line 77 looks for a check of travis-ci.com, that should be enough, we don't need to have a separate error message to signal there are zero checks. as we already have one to signal no "travis-ci.com" checks.